core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.cpp

```diff
@@ -104,16 +104,19 @@ void OpenMPTargetExec::clear_scratch() {
 
 void* OpenMPTargetExec::get_scratch_ptr() { return m_scratch_ptr; }
 
-void OpenMPTargetExec::resize_scratch(int64_t reduce_bytes,
-                                      int64_t team_reduce_bytes,
-                                      int64_t team_shared_bytes,
-                                      int64_t thread_local_bytes) {
+void OpenMPTargetExec::resize_scratch(int64_t team_size, int64_t shmem_size_L0,
+                                      int64_t shmem_size_L1) {
   Kokkos::Experimental::OpenMPTargetSpace space;
-  int64_t total_size =
-      MAX_ACTIVE_TEAMS * reduce_bytes +         // Inter Team Reduction
-      MAX_ACTIVE_TEAMS * team_reduce_bytes +    // Intra Team Reduction
-      MAX_ACTIVE_TEAMS * team_shared_bytes +    // Team Local Scratch
-      MAX_ACTIVE_THREADS * thread_local_bytes;  // Thread Private Scratch
+  const int64_t shmem_size =
+      shmem_size_L0 + shmem_size_L1;  // L0 + L1 scratch memory per team.
+  const int64_t padding = shmem_size * 10 / 100;  // Padding per team.
+  // FIXME_OPENMPTARGET - Total amount of scratch memory allocated is depenedent
+  // on the maximum number of teams possible. Currently the maximum number of
+  // teams possible is calculated based on NVIDIA's Volta GPU and assuming that
+  // the number of threads per threadblock will not exceed 1024 threads. In
+  // future this value should be based on the chosen architecture for the
+  // OpenMPTarget backend.
+  int64_t total_size = (shmem_size + 16 + padding) * ((1024 * 80) / team_size);
```

Comment:
we probably need to build our own list for this, how fun ...

Reply:
The reason I used 1024 instead of 2048 is because it might be very rare that all the threads in a thread block will be used. Do you want me to use 2048 anyway?